src/aac/plugins/gen_json/gen_json_impl.py

```diff
@@ -10,22 +11,51 @@
 )
 from aac.validator import validation
 
+
 plugin_name = "gen-json"
 
 
-def print_json(architecture_files: list[str]) -> PluginExecutionResult:
-    """Print the parsed_models from the parsed architecture_files values in JSON format."""
+def print_json(architecture_files: list[str], output_directory: str = None) -> PluginExecutionResult:
+    """Prints the parsed_models from the parsed architecture_files values in JSON format.
+
+       (optional) if output directory is given to the cli, parsed_models with be writen to
+        a new file in user defined directory
+
+    """
+
+    def to_json(architecture_file: str, output_directory: str) -> str:
+        file_name = os.path.basename(architecture_file)
+        file_ext = os.path.splitext(file_name)[1]
 
-    def to_json(architecture_file: str) -> str:
         with validation(parse_file, architecture_file) as result:
-            return f"File: {architecture_file}\n{json.dumps(result.model, indent=2)}\n"
+
+            if output_directory is None:
+                return f"File: {architecture_file}\n{json.dumps(result.model, indent=2)}\n"
```

Comment:
Since we're using the same `json.dumps(...)` call in two places, now, it is a good candidate for being put into a variable.

Reply:
can this be called within the same line? so it would look like this. 

export = json.dumps(result.model, indent=2)

return f"File: {architecture_file}\n{export}\n"

Reply:
I'm not really sure I understand what you mean by "called within the same line" since the code you provided was two lines. The only thing I'd add to what you put in your comment is that the `formatted_json = json.dumps(...)` part should go outside the `if output_directory` part so it can be used in the `out_file.write(...)` line.

Reply:
my bad, horrible wording. I was wondering if I could call the variable within the return line the same way json.dumps(...) is currently called. By your comment, I see that the answer is yes